process.rb

```diff
@@ -87,17 +87,23 @@ def slugify(word)
 # /_ballots/oakland/2018-11-06.md
 ELECTIONS.each do |election_name, election|
   locality, _time = election_name.split('-', 2)
-  office_elections = OfficeElection.where(election_name: election_name).pluck(:title).uniq
+  office_elections = OfficeElection.where(election_name: election_name)
   referendums = OaklandReferendum.where(election_name: election_name).pluck(:Short_Title).uniq
   ballot_name = "/_ballots/#{locality}/#{election[:date]}.md"
+  office_elections_by_label = office_elections.group_by(&:label)
 
   build_file(ballot_name) do |f|
     f.puts(YAML.dump(
       'title' => election[:title],
       'locality' => locality,
       'election' => election[:date],
-      'office_elections' => office_elections.map do |office|
-        "_office_elections/#{locality}/#{election[:date]}/#{slugify(office)}.md"
+      'office_elections' => office_elections_by_label.map do |label, items|
+        {
+          'label' => label,
+          'items' => items.map do |office_election|
+            "_office_elections/#{locality}/#{election[:date]}/#{slugify(office_election.title)}.md"
+          end
+        }.compact
```

Comment:
I agree, if it was nil, but in this case label is `''` which I don't like because the yaml comes out as:

```
label:
```

Jekyll handles this fine either way, I just prefer nothing to empty string. Is there an easy way to convert empty string to `nil` in Ruby?

Reply:
Not in Ruby, but [ActiveSupport has a method](https://guides.rubyonrails.org/active_support_core_extensions.html#presence):

```
require 'active_support/core_ext/object/blank'

> "foo".presence
"foo"

> "".presence
nil
```

Reply:
I think i'll try that.

I just saw that `csvsql` has a `--blanks` option, but the description seems to be the behavior that we already have. I wonder if there's a bug.

```
--blanks              Do not coerce empty strings to NULL values.
```

Reply:
I switched to `.presence` but still getting empty strings :man_shrugging: